src/tests/createHeadlessForm.test.js

```diff
@@ -1578,6 +1580,89 @@ describe('createHeadlessForm', () => {
       });
     });
 
+    it('supports "null" field type without inputType', () => {
+      const { handleValidation, fields } = createHeadlessForm(schemaNullField, {
+        strictInputType: false,
+      });
+
+      expect(fields).toMatchObject([
+        {
+          name: 'name',
+          label: '(Optional) Name',
+          type: undefined,
+          jsonType: 'null',
+          schema: expect.any(Object),
+        },
+        {
+          name: 'username',
+          label: 'Username',
+          type: 'text',
+          jsonType: 'string',
+          maxLength: 4,
+          schema: expect.any(Object),
+        },
+      ]);
+
+      const validateForm = (vals) => friendlyError(handleValidation(vals));
+
+      expect(validateForm({})).toEqual({
+        username: 'Required field',
+      });
+
+      expect(validateForm({ username: 'hello' })).toEqual({
+        username: 'Please insert up to 4 characters',
+      });
+
+      expect(validateForm({ username: 'john' })).toBeUndefined();
+      expect(validateForm({ name: 2, username: 'john' })).toBeUndefined();
```

Comment:
If you are curious, check jsonschemalint.com, it fails there 

<details>

<img width="1574" alt="image" src="https://github.com/user-attachments/assets/12656fd7-a307-4ee1-9cbc-dfe293732675">
<details>

Reply:
hum, maybe I'm confusing the schema – and its validations – with the presentational aspect of a `type: null` field. From the docs, a `type: null` defaults to a text input, and it currently allows submitting the value of that text input, probably incorrectly though. Thanks for sharing the lint example, I'll update to only accept `null`/absent values.

Reply:
Good catch, so that's a bug or miss-understanding... That made sense until now, but now that we are considering fields that don't carry any value, then it doesn't make sense. We can keep the default to `text` (as we don't have the concept of `hidden` in JSF). But the validation must be fixed to now allow it (that's a bug) 🐛  WDYT?

Reply:
Maybe that's a documentation issue based on a leak of a Dragon abstraction (DynamicForm) into json-schema-form docs?

The output of this field has no `inputType` attribute contrary to what the mapping in the docs suggests:
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/36643065-02af-4fef-b7f8-ae552a61b71a">

So in a way, JSF is not opinionated on what input type should be used for `null` types. Maybe the docs should look like this?

<details><summary>Screenshot</summary>
<p>

<img width="1036" alt="image" src="https://github.com/user-attachments/assets/93264a1c-9a30-47c0-9105-06c3375056db">


</p>
</details> 

> But the validation must be fixed to now allow it (that's a bug) 🐛 WDYT?

Done!

Reply:
Then even better, update the docs for that.